### PR TITLE
Magladko rosdep base qtwebengine5 dev

### PR DESCRIPTION
**Pull request by magladko**
Please add the following dependency to the rosdep database.

## Package name:

qtwebengine5-dev

## Package Upstream Source:

https://github.com/qt/qtwebengine

## Purpose of using this:

Support for Qt5 Webengine utilities to make it possible to work with the Qt5 GUI application project using the webengine utilities.

Distro packaging links:

## Links to Distribution Packages

- Debian: https://packages.debian.org/
  - https://packages.debian.org/bookworm/qtwebengine5-dev
- Ubuntu: https://packages.ubuntu.com/ [available only from focal onwards, added nulls only for current distros]
  - https://packages.ubuntu.com/jammy/qtwebengine5-dev
  - https://packages.ubuntu.com/search?keywords=qtwebengine5-dev&searchon=names&suite=all&section=all
- Fedora: https://packages.fedoraproject.org/
  - https://packages.fedoraproject.org/pkgs/qt5-qtwebengine/qt5-qtwebengine/
- Arch: https://www.archlinux.org/packages/
  - not available
- Gentoo: https://packages.gentoo.org/
  - https://packages.gentoo.org/packages/dev-qt/qtwebengine
- macOS: https://formulae.brew.sh/
  - not available
- Alpine: https://pkgs.alpinelinux.org/packages
  - not available
- NixOS/nixpkgs: https://search.nixos.org/packages
  - https://search.nixos.org/packages?channel=23.11&show=libsForQt5.qt5.qtwebengine&from=0&size=50&sort=relevance&type=packages&query=webengine
- openSUSE: https://software.opensuse.org/package/
  - not available
- rhel: https://rhel.pkgs.org/
  - https://alpine.pkgs.org/3.18/alpine-community-aarch64/qt5-qtwebengine-dev-5.15.13-r13.apk.html

### CODE REVIEW

File: rosdep/base.yaml [hunk 1]
```diff
@@ -7424,6 +7424,21 @@ qttools5-dev-tools:
   openembedded: [qttools@meta-qt5]
   rhel: [qt5-qttools-devel]
   ubuntu: [qttools5-dev-tools]
+qtwebengine5-dev:
+  alpine: null
+  arch: null
```
Comment (clalancette) [inline rosdep/base.yaml:7429]: No need to add nulls for entries we don't have:

```suggestion
```

File: rosdep/base.yaml [hunk 2]
```diff
@@ -7424,6 +7424,21 @@ qttools5-dev-tools:
...
+  debian: [qtwebengine5-dev]
+  fedora: [qt5-qtwebengine]
+  gentoo: [dev-qt/qtwebengine-5.15.11_p20231120]
+  nixos: [libsForQt5.qt5.qtwebengine]
+  opensuse: null
+  osx: null
```
Comment (clalancette) [inline rosdep/base.yaml:7435]: Same:

```suggestion
```

File: rosdep/base.yaml [hunk 3]
```diff
@@ -7424,6 +7424,21 @@ qttools5-dev-tools:
...
+  rhel: qt5-qtwebengine-dev
```
Comment (clalancette) [inline rosdep/base.yaml:7436]: This needs to be a list:

```suggestion
  rhel: [qt5-qtwebengine-devel]
```

File: rosdep/base.yaml [hunk 4]
```diff
@@ -7424,6 +7424,21 @@ qttools5-dev-tools:
   openembedded: [qttools@meta-qt5]
   rhel: [qt5-qttools-devel]
   ubuntu: [qttools5-dev-tools]
+qtwebengine5-dev:
+  alpine: null
+  arch: null
+  debian: [qtwebengine5-dev]
+  fedora: [qt5-qtwebengine]
```
Comment (clalancette) [inline rosdep/base.yaml:7431]:
```suggestion
  fedora: [qt5-qtwebengine-devel]
```

File: rosdep/base.yaml [hunk 5]
```diff
@@ -7424,6 +7424,21 @@ qttools5-dev-tools:
...
+  gentoo: [dev-qt/qtwebengine-5.15.11_p20231120]
+  nixos: [libsForQt5.qt5.qtwebengine]
+  opensuse: null
+  osx: null
+  rhel: qt5-qtwebengine-dev
+  ubuntu: 
+    '*': [qtwebengine5-dev]
+    bionic: null
+    trusty: null
+    xenial: null
```
Comment (clalancette) [inline rosdep/base.yaml:7441]: No need to add in keys for end-of-life distributions:

```suggestion
  ubuntu: [qtwebengine5-dev]
```

**Review by clalancette — CHANGES_REQUESTED** [review]

**Comment by magladko** [comment]
@clalancette thanks for the changes, it's my first PR here

File: rosdep/base.yaml [hunk 6]
```diff
@@ -7424,6 +7424,13 @@ qttools5-dev-tools:
   openembedded: [qttools@meta-qt5]
   rhel: [qt5-qttools-devel]
   ubuntu: [qttools5-dev-tools]
+qtwebengine5-dev:
+  debian: [qtwebengine5-dev]
+  fedora: [qt5-qtwebengine-devel]
+  gentoo: [dev-qt/qtwebengine-5.15.11_p20231120]
```
Comment (clalancette) [inline rosdep/base.yaml:7430]:
```suggestion
  gentoo: [dev-qt/qtwebengine]
```

**Review by clalancette — CHANGES_REQUESTED** [review]

**Review by clalancette — APPROVED** [review]

**Review by mjcarroll — APPROVED** [review]

**Pull request merged**